### PR TITLE
Add windows_sys type definitions for ARM32 manually

### LINKED ISSUE #112265: windows-sys generated WinAPI bindings break ARM platforms
In https://github.com/rust-lang/rust/pull/110152 the WinAPI definitions have been replaced by windows-bindgen generated bindings. However, the [win32metadata](https://github.com/microsoft/win32metadata) project does not generate metadata for ARM32-specific items, hence the generated bindings lack a few types like `CONTEXT` and `WSADATA` on this platform. It effectively breaks `thumbv7a-*-windows-*` for Rust.

@ChrisDenton

### PR DESCRIPTION
**Pull request by bdbai**
`windows_sys` bindings do not include platform-specific type definitions for ARM 32 architecture, so it breaks `thumbv7a-pc-windows-msvc` and `thumbv7a-uwp-windows-msvc` targets. This PR tries to add them back by manually inserting them together with the generated ones.

`WSADATA` is copied from the generated definition for `x86`. `CONTEXT` is copied from the definition before `windows_sys` is introduced (which is just an empty enum): https://github.com/rust-lang/rust/blob/4a18324a4df6bc98bec0b54d35908d7a9cdc7c32/library/std/src/sys/windows/c.rs#L802.

Fixes #112265.

### CODE REVIEW

File: src/tools/generate-windows-sys/src/arm_shim.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Begin of ARM32 shim
+cfg_if::cfg_if! {
+if #[cfg(target_arch = "arm")] {
```
Comment (faptc) [inline src/tools/generate-windows-sys/src/arm_shim.rs:3]: The `if` only covers one target_arch, maybeb you don't need cfg_if at all.

Comment (bdbai) [reply src/tools/generate-windows-sys/src/arm_shim.rs:3]: I just don't want to repeat the cfg guard twice (or even more in the future).

**Comment by ChrisDenton** [comment]
This looks good to me. I might want to think more about how to organise platform support but let's fix this issue first.

Could you add a short comment to `arm_shim.rs` explaining how it's included? Nothing long but it's not part of the normal module system so I'd like to make that clear for anyone reading.

**Comment by bdbai** [comment]
> Could you add a short comment to `arm_shim.rs` explaining how it's included? Nothing long but it's not part of the normal module system so I'd like to make that clear for anyone reading.

@ChrisDenton I have added some comments in https://github.com/rust-lang/rust/pull/112527/commits/df08f56b089927fd5552b208d4c0a6cd3b8f12b0. Does it makes sense?

**Comment by ChrisDenton** [comment]
Yep makes sense to me, thanks!

@bors r+ rollup

**Pull request merged**